container_linux.go

```diff
@@ -213,6 +215,12 @@ func newPipe() (parent *os.File, child *os.File, err error) {
 func (c *linuxContainer) Destroy() error {
 	c.m.Lock()
 	defer c.m.Unlock()
+	// Since the state.json and CRIU image files are in the c.root
+	// directory, we should not remove it after checkpoint.  Also,
+	// when     CRIU exits after restore, we should not kill the processes.
+	if _, err := os.Stat(filepath.Join(c.root, "checkpoint")); err == nil {
```

Comment:
@LK4D4 @crosbymichael 
For Docker containers, I use "criu_img" subdirectory in "/var/lib/containers/<ID>" which is outside the container's root.  I think @crosbymichael used the "checkpoint" subdirectory in "/nsinit" of the container because nsinit is really a test program.  In general, I agree with you that we do not want to save the files that CRIU generates (images, logs, and pid) inside container's root.